src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/storm/bolt/speaker/SpeakerRouter.java

```diff
@@ -97,12 +102,20 @@
     public static final String STREAM_BFD_WORKER_ID = "worker";
     public static final Fields STREAM_BFD_WORKER_FIELDS = new Fields(FIELD_ID_KEY, FIELD_ID_INPUT, FIELD_ID_CONTEXT);
 
+    public static final String STREAM_ZOOKEEPER_ID = ZkStreams.ZK.toString();
+    public static final Fields STREAM_ZOOKEEPER_FIELDS = new Fields(ZooKeeperBolt.FIELD_ID_STATE,
+            ZooKeeperBolt.FIELD_ID_CONTEXT);
+
     @Override
     protected void handleInput(Tuple input) throws Exception {
         String source = input.getSourceComponent();
         if (ComponentId.INPUT_SPEAKER.toString().equals(source)) {
             Message message = pullValue(input, FIELD_ID_INPUT, Message.class);
             speakerMessage(input, message);
+        } else if (ComponentId.INPUT_ZOOKEEPER.toString().equals(source)) {
+            LifecycleEvent event = (LifecycleEvent) input.getValueByField(FIELD_ID_LIFECYCLE_EVENT);
+            handleLifeCycleEvent(event);
+            emit(STREAM_ZOOKEEPER_ID, input, new Values(event, getCommandContext()));
```

Comment:
I think we do not need `emit()` here because we already send tuple to ZkBolt in `handleLifeCycleEvent()`

Reply:
Removed.